common-utils/src/main/java/eu/arrowhead/common/mqtt/ArrowheadMqttService.java

```diff
@@ -0,0 +1,112 @@
+package eu.arrowhead.common.mqtt;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.eclipse.paho.client.mqttv3.MqttClient;
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.eclipse.paho.client.mqttv3.MqttMessage;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Service;
+import org.springframework.util.Assert;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import eu.arrowhead.common.Constants;
+import eu.arrowhead.common.Utilities;
+import eu.arrowhead.common.exception.ExternalServerError;
+import eu.arrowhead.common.exception.InternalServerError;
+import eu.arrowhead.dto.MqttResponseTemplate;
+
+@Service
+@ConditionalOnProperty(name = Constants.MQTT_API_ENABLED, matchIfMissing = false)
+public class ArrowheadMqttService {
+
+	//=================================================================================================
+	// members
+
+	@Autowired
+	private MqttService mqttService;
+
+	@Autowired
+	private ObjectMapper mapper;
+
+	private final Logger logger = LogManager.getLogger(getClass());
+
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------
+	/**
+	 * Subscribe for consuming a push service
+	 */
+	public void subscribe(final String connectId, final String topic, final MqttQoS qos) {
+		logger.debug("subscribe started");
+		Assert.isTrue(!Utilities.isEmpty(connectId), "connectId is empty");
+		Assert.isTrue(!Utilities.isEmpty(topic), "topic is empty");
+
+		// TODO
```

Comment:
it would be wise to create an issue about this TODO, just to not forget to implement it.